### PR TITLE
support JSONBody

### PR DESCRIPTION
**Pull request by rogpeppe**
We allow a body to be specified as a Go value, a common requirement.

(Review request: http://reviews.vapour.ws/r/1442/)

### CODE REVIEW

File: httptesting/http.go [hunk 1]
```diff
@@ -146,6 +152,12 @@ type DoRequestParams struct {
 	// URL holds the URL to pass when making the request.
 	URL string
 
+	// JSONBody specifies a JSON value to marshal to use
+	// as the body of the request. If this is specified, Body will
+	// be ignored and an appropriate Content-Type header will
```
Comment (mhilton) [inline httptesting/http.go]: Why not just say Content-Type will be set to "application/json"?

Comment (rogpeppe) [inline httptesting/http.go]: done

**Comment by mhilton** [comment]
:+1: LGTM

File: httptesting/http.go [hunk 2]
```diff
@@ -180,10 +193,18 @@ func DoRequest(c *gc.C, p DoRequestParams) *httptest.ResponseRecorder {
 	srv := httptest.NewServer(p.Handler)
 	defer srv.Close()
 
+	if p.JSONBody != nil {
+		data, err := json.Marshal(p.JSONBody)
+		c.Assert(err, gc.IsNil)
```
Comment (mattyw) [inline httptesting/http.go]: I'll let this slide this time

**Comment by mattyw** [comment]
LGTM

File: httptesting/http.go (hunk 2 above)
Comment (rogpeppe) [inline httptesting/http.go]: done anyway

**Comment by rogpeppe** [comment]
$$merge$$

**Pull request merged**